### ISSUE TITLE
Cannot listen to any port inside container

### ISSUE THREAD

**Issue by soanvig** [docker/for-linux#706]
* [x] This is a bug report
* [ ] This is a feature request
* [ ] I searched existing issues before opening this one

### Expected behavior

I can start any application on any "classic" (3000, 5432, 8080, 8000, any) port inside docker container.

### Actual behavior

I cannot start **any** port listening inside my docker containers. My node.js apps, mysql or postgres apps. I get `permission denied` kindof errors (it depends on the engine of the app).

Sample error: `httpd: socket: permission denied` (port 8080).
or: `LOG: could not create IPv4 socket for address "0.0.0.0": Permission denied" (port 5432)

Running as sudo doesn't help

### Steps to reproduce the behavior

No exposing ports, nothing. Just app wants to listen on port inside container.

`docker run postgres:alpine`

`docker run -e MYSQL_ROOT_PASSWORD="foobar" mysql`

`docker run crccheck/hello-world`

**Output of `docker version`:**

```
Client:
 Version:           18.09.6
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        481bc77
 Built:             Sat May  4 02:36:00 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.6
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       481bc77
  Built:            Sat May  4 01:59:36 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Output of `docker info`:**

```
Containers: 17
 Running: 0
 Paused: 0
 Stopped: 17
Images: 7
Server Version: 18.09.6
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: bb71b10fd8f58240ca47fbb579b9d1028eea7c84
runc version: 2b18fe1d885ee5083ef9f0838fee39b62d653e30
init version: fec3683
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-30deepin-generic
Operating System: Deepin 15
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.41GiB
Name: mortimer-carbon
ID: LM6Y:SGPK:6JLK:7ETY:74A6:3FHL:PHT3:I3IT:VKRN:SQGY:5ZTO:CEH4
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine

```

**Additional environment details (AWS, VirtualBox, physical, etc.)**

Physical, notebook, fresh install, linux debian-based 64bit

**Comment by drwatson12** [comment]
Hi @soanvig  - Did this problem get resolved? We're having the same problem as well with Inter Container communication post RHEL 7.4 to 7.6 upgrade.

**Comment by soanvig** [comment]
Nope, I've changed the system from debian-based to arch-based, but it is only a coincidence. I couldn't find solution. Docker on the exact same system, on another PC works flawlessly. I don't know what is the problem.

My notebook I had problem with Docker on was Lenovo Carbon X1 gen6, but I doubt it is hardware related.

**Comment by drwatson12** [comment]
So even after changing from Debian-based to arch-based, the problem is unresolved. 
Anyway, we are working with Docker support on this, so will keep you posted.

**Comment by soanvig** [comment]
No, switching to another system "fixed" the problem, I can use docker now. But it is not "fix", it's workaround.

**Comment by Eyshika** [comment]
@soanvig Is there any way to resolve it in being linux only ? I have almost same problem, I have a website running on 8097 port of container which I want to access from host ip :<any port> , for that i did `docker -run --publish=8098:8098 <container-name>` but it started new container. i wanted to be on running container since I have all the data stored there

**Comment by soanvig** [comment]
Sorry, I haven't found the solution

**Comment by arkodg** [comment]
There have been issues with Docker on Debian recently, this comment should help
https://github.com/moby/moby/issues/38974#issuecomment-504148014

**Comment by abcfy2** [comment]
I have the same issue, but when I remove `apparmor`, it's working.

**Comment by qwe852** [comment]
> I have the same issue, but when I remove `apparmor`, it's working.

Do you means remove the `arrarmor` on host?
